#!/usr/bin/env python3
# encoding: utf-8

from typing import Final

PACKAGE: Final[str] = 'catkin_template'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("int_param",    int_t,    0, "integer parameter",  50,  0, 100)
gen.add("double_param", double_t, 0, "double parameter",  0.5,  0,   1)
gen.add("str_param",    str_t,    0, "string parameter",  "Hello World")
gen.add("bool_param",   bool_t,   0, "boolean parameter", True)

size_enum = gen.enum([ gen.const("S",  int_t, 0, "small constant"),
                       gen.const("M",  int_t, 1, "medium constant"),
                       gen.const("L",  int_t, 2, "large constant"),
                       gen.const("XL", int_t, 3, "extra large constant")],
                     "enum to set size")

gen.add("size", int_t, 0, "enum-size parameter", 1, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, f"{PACKAGE}_cfg", "Config"))